rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -2987,6 +3020,87 @@ extern "C" rmw_ret_t rmw_get_node_names(
   return RMW_RET_BAD_ALLOC;
 }
 
+extern "C" rmw_ret_t rmw_get_node_names_with_security_contexts(
+  const rmw_node_t * node,
+  rcutils_string_array_t * node_names,
+  rcutils_string_array_t * node_namespaces,
+  rcutils_string_array_t * security_contexts)
+{
+  RET_WRONG_IMPLID(node);
+  auto node_impl = static_cast<CddsNode *>(node->data);
+  if (rmw_check_zero_rmw_string_array(node_names) != RMW_RET_OK ||
+    rmw_check_zero_rmw_string_array(node_namespaces) != RMW_RET_OK)
+  {
+    return RMW_RET_ERROR;
+  }
+
+  std::vector<std::tuple<std::string, std::string, std::string>> ns;
+  const auto re = std::regex(
+    "^name=([^;]*);namespace=([^;]*);securitycontext=([^;]*);", std::regex::extended);
```

Comment:
@ivanpauno nit: why not calling the constructor directly i.e. `std::regex re(...);`?

Reply:
preexisting, I can address it if you mind strongly.

P.S.: As file as I understand, this is calling just the constructor.

Reply:
Yeah, copy elision will eventually do the job but it's odd. OK to leave as-is if pre-existing.

Reply:
Don't feel like you have to leave non-idiomatic C++ bits untouched. Chances are that I wrote them, and as I'm more used to C than to C++ and have grown to dislike both, my C++ tends to come a bit weird sometimes.

Reply:
Done in https://github.com/ros2/rmw_cyclonedds/pull/106/commits/3c2f89c4414f5f9df88b3bc7d71197b0a069152d.